include/ignition/gui/GuiEvents.hh

```diff
@@ -255,6 +255,56 @@ namespace ignition
         /// for this event.
         private: bool menuEnabled;
       };
+
+      /// \brief Event which is called to broadcast the key release within
+      /// the scene.
+      class KeyRelease : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _key The released key within the scene
+        public: explicit KeyRelease(const int &_key)
+            : QEvent(kType), key(_key)
+        {
+        }
+
+        /// \brief Unique type for this event.
+        static const QEvent::Type kType = QEvent::Type(QEvent::MaxUser - 8);
+
+        /// \brief Get the released key within the scene that the user released.
+        /// \return The key code.
+        public: int Key() const
+        {
+          return this->key;
+        }
+
+        /// \brief The key that the user released within the scene.
+        private: int key;
+      };
+
+      /// \brief Event which is called to broadcast the key press within
+      /// the scene.
+      class KeyPress : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _key The pressed key within the scene
+        public: explicit KeyPress(const int &_key)
+            : QEvent(kType), key(_key)
+        {
+        }
+
+        /// \brief Unique type for this event.
+        static const QEvent::Type kType = QEvent::Type(QEvent::MaxUser - 9);
```

Comment:
Where does this `-9` come from?  Seems kind of magic.

Reply:
The first event type starts with `0`, then we are subtracting one to each new event

Render:
https://github.com/ignitionrobotics/ign-gui/blob/a983a6d7e39473ee69ebf505b7b82f04b8d7c605/include/ignition/gui/GuiEvents.hh#L45

SnapIntervals:
https://github.com/ignitionrobotics/ign-gui/blob/a983a6d7e39473ee69ebf505b7b82f04b8d7c605/include/ignition/gui/GuiEvents.hh#L87